Sources/StreamChat/Repositories/AuthenticationRepository.swift

```diff
@@ -275,7 +265,7 @@ class AuthenticationRepository {
     }
 
     private func updateToken(token: Token?, notifyTokenWaiters: Bool) {
-        let waiters: [String: (Result<Token, Error>) -> Void] = tokenQueue.sync {
+        let waiters: [String: (Result<Token, Error>) -> Void] = tokenQueue.sync(flags: .barrier) {
```

Comment:
This must be barrier because the work item mutates state (our queue is concurrent, with serial it would be fine)